### ISSUE TITLE
Accuracy

### ISSUE THREAD

**Issue by pexmar** [vsl9/Sentiment-Analysis-with-Convolutional-Networks#6]
Hi,
which accuracy did the Yoon Kim CNN achieve with these settings?

**Comment by vsl9** [comment]
Hi,

This project was started in 2015. At that time Keras supported Sequential models only, so it wasn't possible to reproduce Yoon Kim's model in a straightforward manner.

I've tested Kim's model via Model functional API in Keras 1.2.0:
```python
# Number of feature maps (outputs of convolutional layer)
N_fm = 100
# kernel size of convolutional layer
kernel_size = [3, 4, 5]

model_input = Input(shape=(train_X.shape[-1],))
# Embedding layer (lookup table of trainable word vectors)
model_output = Embedding(input_dim=W.shape[0], 
                         output_dim=W.shape[1], 
                         input_length=conv_input_height,
                         weights=[W], 
                         W_constraint=unitnorm())(model_input)
# Reshape word vectors from Embedding to tensor format suitable for Convolutional layer
model_output = Reshape((1, conv_input_height, conv_input_width))(model_output)

conv_outputs = []
for ks in kernel_size:
    # first convolutional layer
    conv_output = Convolution2D(N_fm, 
                                ks, 
                                conv_input_width, 
                                border_mode='valid', 
                                W_regularizer=l2(0.0001))(model_output)
    # ReLU activation
    conv_output = Activation('relu')(conv_output)
    # aggregate data in every feature map to scalar using MAX operation
    conv_output = MaxPooling2D(pool_size=(conv_input_height-ks+1, 1))(conv_output)
    conv_output = Flatten()(conv_output)
    conv_outputs.append(conv_output)

model_output = Merge(mode='concat')(conv_outputs)
model_output = Dropout(0.5)(model_output)
# Inner Product layer (as in regular neural network, but without non-linear activation function)
model_output = Dense(2)(model_output)
# SoftMax activation; actually, Dense+SoftMax works as Multinomial Logistic Regression
model_output = Activation('softmax')(model_output)
model = Model(model_input, model_output)
```
After 3 epochs of training:
* validation accuracy is 85.11%, validation AUC is 92.55% for Kim's model
* validation accuracy is 87.41%, validation AUC is 94.53% for my model
